src/main/java/org/jboss/logmanager/handlers/WriterHandler.java

```diff
@@ -59,25 +73,66 @@ protected void doPublish(final ExtLogRecord record) {
             return;
         }
         try {
-            lock.lock();
+            write(record, formatted);
+        } catch (Exception ex) {
+            reportError("Error writing log message", ex, ErrorManager.WRITE_FAILURE);
+        }
+    }
+
+    private void lockedWrite(ExtLogRecord record, String formatted) throws IOException {
+        lock.lock();
+        try {
+            doWrite(record, formatted);
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    private boolean doWrite(ExtLogRecord record, String formatted) throws IOException {
+        assert lock.isHeldByCurrentThread();
+        if (writer == null) {
+            return true;
+        }
+        preWrite(record);
+        final Writer writer = this.writer;
+        if (writer == null) {
+            return true;
+        }
```

Comment:
Is there a reason for this? It's only used in the lock.

Reply:
Same here, the code was refactored from ye original code